h2/src/main/org/h2/engine/Session.java

```diff
@@ -226,10 +229,25 @@ public SubQueryInfo getSubQueryInfo() {
         return subQueryInfo;
     }
 
-    public void setParsingView(boolean parsingView) {
+    public void setParsingView(boolean parsingView, String viewName) {
         // It can be recursive, thus implemented as counter.
         this.parsingView += parsingView ? 1 : -1;
         assert this.parsingView >= 0;
+        if(parsingView){
+            viewNameStack.push(viewName);
+        }else
+        {
+            assert viewName.equals(viewNameStack.peek());
+            viewNameStack.pop();
+        }        
+    }
+    public String getParsingViewName() {
+        try{
+        return viewNameStack.peek();
+        }
+        catch(EmptyStackException e){
```

Comment:
rather check the size of the stack than catch an exception, that's unneccessarily expensive